src/main/java/com/intuit/graphql/orchestrator/federation/exceptions/FederationDirectiveInvalidProviderException.java

```diff
@@ -1,12 +1,12 @@
-package com.intuit.graphql.orchestrator.federation.keydirective.exceptions;
+package com.intuit.graphql.orchestrator.federation.exceptions;
 
 import com.intuit.graphql.orchestrator.stitching.StitchingException;
 
-public class InvalidLocationForFederationDirective extends StitchingException {
+public class FederationDirectiveInvalidProviderException extends StitchingException {
```

Comment:
I suggest add a javadoc for the purpose/usage of the exception.  I like to old name better, but my suggestion is rename to `InvalidUseOfFederationDirective`.   Let's reserve the `Location` word for errors that don't satisfy the directive location like  in`directive @requires(fields: _FieldSet!) on FIELD_DEFINITION`.  But should be catch by the parser already.

Another suggestion is:
`InvalidUseOfFederationSchemaElement` - this can be mapped to an error code on the app server side then extended by `InvalidUseOfFederationDirective`

Reply:
@scanbns might have input on exception structure from the perspective of error code mappings

Reply:
Removing Exception entirely after talking to Ashpak

Reply:
Wants to just skip the logic of code entirely if it is not a federation provider